### ISSUE TITLE
Proposal: Rename some artifacts

### ISSUE THREAD

**Issue by shayne-fletcher** [digital-asset/ghc-lib#406]
# Proposal

## tl;dr Rename some artifacts
e.g. `mini-hlint to ghc-lib-test-mini-hlint`

## Details
The `ghc-lib` project produces a bunch of packages. Each package is described a distinct .cabal file:

  - ghc-lib-gen
  - ghc-lib-parser
  - ghc-lib
  - test-utils
  - mini-hlint
  - mini-compile

The proposal is to rename the last three targets as per the following table.

before | after
--- | ---
test-utils  | ghc-lib-test-utils
mini-hlint  | ghc-lib-test-mini-hlint
mini-compile | ghc-lib-test-mini-compile

The motivation is to make these targets more readily identifiable as `ghc-lib` project artifacts. This makes reading build logs of the `hlint` stack easier.

After the renamings, the package list will read:

  - ghc-lib-gen
  - ghc-lib-parser
  - ghc-lib
  - ghc-lib-test-utils
  - ghc-lib-test-mini-hlint
  - ghc-lib-test-mini-compile

**Comment by shayne-fletcher** [comment]
@garyverhaegen-da

**Comment by garyverhaegen-da** [comment]
Makes sense to me. Thanks!
